### PR TITLE
Issue 245: Cache the factory function symbol for records using the cl…

### PR DESCRIPTION
**Pull request by WilliamParker**
…ass as the key to improve performance of session serialization

### CODE REVIEW

File: src/main/clojure/clara/rules/durability/fressian.clj [hunk 1]
```diff
@@ -39,23 +39,34 @@
...
-        idx (.lastIndexOf class-name (int \.))
-        ns-nom (.substring class-name 0 idx)
-        nom (.substring class-name (inc idx))]
-    ;; There could be some sort of cache for these based on class.  It may speed
-    ;; up serialization, but doesn't seem worth it yet.
-    (symbol (str (cm/demunge ns-nom)
-                 "/map->"
-                 (cm/demunge nom)))))
+  (let [klass (class rec)]
+    (if-let [cached-sym (.get ^Map class->factory-fn-sym klass)]
```
Comment (mrrodriguez) [inline src/main/clojure/clara/rules/durability/fressian.clj]: nit: if you just type hint the `def` you wouldn't have to type hint it for all references to it.
```clj
(def ^:private ^Map class->factory-fn-sym (WeakHashMap.))
```

**Comment by mrrodriguez** [comment]
+1 other than my minor comment.

**Comment by WilliamParker** [comment]
@mrrodriguez I have moved the type hint and updated the pull request.

**Comment by mrrodriguez** [comment]
@wparker looks good to me

**Pull request merged**